tests/test_user_lock.rs

```diff
@@ -0,0 +1,63 @@
+use colink::{
```

Comment:
filename: actually, although we use a prefix `user_` to distinguish examples, I'm not sure if we still need to do that for tests. Considering it is fair to call this a test for lock, maybe we should remove the prefix? (same for the func name, same for the other file)

Reply:
Also, it seems that this is a test using a counter, maybe `test_counter_with_lock` is more appropriate?